Sources/Build/Triple.swift

```diff
@@ -132,47 +134,22 @@ public struct Triple: Encodable {
     }
 
     public static let macOS = try! Triple("x86_64-apple-macosx")
-    public static let x86_64Linux = try! Triple("x86_64-unknown-linux-gnu")
-    public static let i686Linux = try! Triple("i686-unknown-linux")
-    public static let ppc64leLinux = try! Triple("powerpc64le-unknown-linux")
-    public static let s390xLinux = try! Triple("s390x-unknown-linux")
-    public static let arm64Linux = try! Triple("aarch64-unknown-linux-gnu")
-    public static let armLinux = try! Triple("armv7-unknown-linux-gnueabihf")
-    public static let armAndroid = try! Triple("armv7a-unknown-linux-androideabi")
-    public static let arm64Android = try! Triple("aarch64-unknown-linux-android")
-    public static let x86_64Android = try! Triple("x86_64-unknown-linux-android")
-    public static let i686Android = try! Triple("i686-unknown-linux-android")
-    public static let windows = try! Triple("x86_64-unknown-windows-msvc")
-
-  #if os(macOS)
-    public static let hostTriple: Triple = .macOS
-  #elseif os(Windows)
-    public static let hostTriple: Triple = .windows
-  #elseif os(Linux)
-    #if arch(x86_64)
-      public static let hostTriple: Triple = .x86_64Linux
-    #elseif arch(i386)
-      public static let hostTriple: Triple = .i686Linux
-    #elseif arch(powerpc64le)
-      public static let hostTriple: Triple = .ppc64leLinux
-    #elseif arch(s390x)
-      public static let hostTriple: Triple = .s390xLinux
-    #elseif arch(arm64)
-      public static let hostTriple: Triple = .arm64Linux
-    #elseif arch(arm)
-      public static let hostTriple: Triple = .armLinux    
-    #endif
-  #elseif os(Android)
-    #if arch(arm)
-      public static let hostTriple: Triple = .armAndroid
-    #elseif arch(arm64)
-      public static let hostTriple: Triple = .arm64Android
-    #elseif arch(x86_64)
-      public static let hostTriple: Triple = .x86_64Android
-    #elseif arch(i386)
-      public static let hostTriple: Triple = .i686Android
-    #endif
-  #endif
+
+    public static func getHostTriple(usingSwiftCompiler swiftCompiler: AbsolutePath) -> Triple {
+        do {
+            let result = try Process.popen(args: swiftCompiler.pathString, "-print-target-info")
+            let output = try result.utf8Output().spm_chomp()
+            let targetInfo = try JSON(string: output)
+            let tripleString: String = try targetInfo.get("target").get("unversionedTriple")
+            return try Triple(tripleString)
+        } catch {
+          #if os(macOS)
+            return .macOS
```

Comment:
It seems unnecessary to special-case macOS here. This really should not ever fail; I think we should just propagate the error (unless you know for sure of a particular case where this can fail in a _normal_ scenario?).

Reply:
It fails during development on macOS with the latest Xcode, as the version of the Swift compiler that comes with it still doesn't have the `-print-target-info` option. Perhaps we should put a TODO to remove the special casing when the version of Xcode with 5.2 comes out.

Reply:
Yeah, we want to maintain self hosting capabilities with at least the latest GM Xcode. This is ensured using the self hosting bots on Swift CI.